transports/webrtc-websys/src/upgrade.rs

```diff
@@ -1,141 +1,115 @@
+    };
+
+    let algo: js_sys::Object = Object::new();
+    Reflect::set(&algo, &"name".into(), &"ECDSA".into()).unwrap();
+    Reflect::set(&algo, &"namedCurve".into(), &"P-256".into()).unwrap();
+    Reflect::set(&algo, &"hash".into(), &hash.into()).unwrap();
+
+    let certificate_promise = RtcPeerConnection::generate_certificate_with_object(&algo)
+        .expect("certificate to be valid");
+
+    let certificate = JsFuture::from(certificate_promise).await?; // Needs to be Send
+
+    let ice: js_sys::Object = Object::new();
+    Reflect::set(&ice, &"urls".into(), &"stun:stun.l.google.com:19302".into()).unwrap();
+
+    let mut config = RtcConfiguration::default();
+    // wrap certificate in a js Array first before adding it to the config object
+    let certificate_arr = js_sys::Array::new();
+    certificate_arr.push(&certificate);
+    config.certificates(&certificate_arr);
+
+    let peer_connection = web_sys::RtcPeerConnection::new_with_configuration(&config)?;
+
+    // Create substream for Noise handshake
+    // Must create data channel before Offer is created for it to be included in the SDP
+    let handshake_data_channel = DataChannel::new_handshake(&peer_connection);
+
+    let webrtc_stream = Stream::new(handshake_data_channel);
+
+    let ufrag = format!("libp2p+webrtc+v1/{}", gen_ufrag(32));
+    /*
+     * OFFER
+     */
+    let offer = JsFuture::from(peer_connection.create_offer()).await?; // Needs to be Send
+    let offer_obj = sdp::offer(offer, &ufrag);
+    log::debug!("Offer SDP: {:?}", offer_obj);
+    let sld_promise = peer_connection.set_local_description(&offer_obj);
+    JsFuture::from(sld_promise)
+        .await
+        .expect("set_local_description to succeed");
+
+    /*
+     * ANSWER
+     */
+    // TODO: Update SDP Answer format for Browser WebRTC
+    let answer_obj = sdp::answer(sock_addr, &remote_fingerprint, &ufrag);
+    log::debug!("Answer SDP: {:?}", answer_obj);
+    let srd_promise = peer_connection.set_remote_description(&answer_obj);
+    JsFuture::from(srd_promise)
+        .await
+        .expect("set_remote_description to succeed");
+
+    // get local_fingerprint from local RtcPeerConnection peer_connection certificate
+    let local_sdp = match &peer_connection.local_description() {
+        Some(description) => description.sdp(),
+        None => return Err(Error::JsError("local_description is None".to_string())),
+    };
+    let local_fingerprint = match sdp::fingerprint(&local_sdp) {
+        Ok(fingerprint) => fingerprint,
+        Err(e) => return Err(Error::JsError(format!("local fingerprint error: {}", e))),
+    };
+
+    log::debug!("local_fingerprint: {:?}", local_fingerprint);
+    log::debug!("remote_fingerprint: {:?}", remote_fingerprint);
+
+    let peer_id = noise::outbound(
+        id_keys,
+        webrtc_stream,
+        remote_fingerprint,
+        local_fingerprint,
+    )
+    .await?;
+
+    log::debug!("peer_id: {:?}", peer_id);
```

Comment:
Nit: If we intend to keep those, we could make them a bit more descriptive.

```suggestion
    log::debug!("Remote peer identified as {peer_id}");
```